Code/Framework/AzFramework/AzFramework/Device/DeviceAttributesSystemComponent.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzFramework/Device/DeviceAttributesSystemComponent.h>
+#include <AzFramework/Device/DeviceAttributes.h>
+
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzCore/Serialization/EditContext.h>
+
+namespace AzFramework
+{
+    DeviceAttributesSystemComponent::DeviceAttributesSystemComponent() = default;
+    DeviceAttributesSystemComponent::~DeviceAttributesSystemComponent() = default;
+
+    void DeviceAttributesSystemComponent::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<DeviceAttributesSystemComponent, AZ::Component>();
+
+            if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+            {
+                editContext->Class<DeviceAttributesSystemComponent>(
+                    "AzFramework Device Attributes Component", "System component responsible for handling device attribute registration")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                    ->Attribute(AZ::Edit::Attributes::Category, "Editor")
+                    ;
+            }
+        }
+    }
+
+    void DeviceAttributesSystemComponent::Activate()
+    {
+        if (DeviceAttributeRegistrar::Get() == nullptr)
+        {
+            DeviceAttributeRegistrar::Register(this);
+        }
+
+        // register default attributes
+        RegisterDeviceAttribute(AZStd::make_unique<DeviceAttributeDeviceModel>());
+        RegisterDeviceAttribute(AZStd::make_unique<DeviceAttributeRAM>());
+    }
+
+    void DeviceAttributesSystemComponent::Deactivate()
+    {
+        if (DeviceAttributeRegistrar::Get() == this)
+        {
+            DeviceAttributeRegistrar::Unregister(this);
+        }
+
+        m_deviceAttributes.clear();
+    }
+
+    bool DeviceAttributesSystemComponent::RegisterDeviceAttribute(AZStd::unique_ptr<DeviceAttribute> deviceAttribute)
+    {
+        auto name = deviceAttribute->GetName();
+        if (m_deviceAttributes.contains(name))
+        {
+            AZ_Warning(
+                "DeviceAttributesSystemComponent", false,
+                "Device attribute '%.*s' is already registered, ignoring new registration request.",
+                AZ_STRING_ARG(name));
+            return false;
+        }
+
+        m_deviceAttributes.emplace(name, AZStd::move(deviceAttribute) );
+        return true;
+    }
+
+    void DeviceAttributesSystemComponent::VisitDeviceAttributes(const VisitInterfaceCallback& callback) const
+    {
+        for (const auto& [deviceAttribue, deviceAttributeInterface] : m_deviceAttributes)

```

Comment:
typo: deviceAttribue
```suggestion
        for (const auto& [deviceAttribute, deviceAttributeInterface] : m_deviceAttributes)

```